### PR TITLE
[CQT-136] Update README

### PR DESCRIPTION
**Pull request by rturrado**
This task is part of a greater task of updating the documentation for the whole of the cQASM tooling.

README files have to be updated. Some information needs to be added, modified, deleted, or moved to other documentation hosting systems (e.g., GitHub pages, ReadTheDocs, or Confluence).

The README files across the cQASM tooling should look consistent.

TODO: the Deployment section in this README has to go to a Confluence page.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,12 +2,53 @@
...
+### Using the Docker Container
+
+The documentation can also be viewed via the provisioned Docker container.
+Run the docker container with the following command
+
+```shell
+docker compose up -d
+```
+
+The documentation can now be viewed at <http://localhost:8106>.
```
Comment (elenbaasc) [inline README.md:44]: _Deployment details should indeed be put in a Confluence page.
This section can be replaced by:_

## Generate the cQASM language specification locally

Clone the repository to your local machine

```shell
git clone https://github.com/QuTech-Delft/cQASM-spec.git
```

It is advised to create a virtual environment (_e.g._, [venv](https://docs.python.org/3/library/venv.html)), run it, and install the requirements

```shell
pip install -r requirements.txt
```

Once the required dependencies have installed, the language specification can be served locally at <http://localhost:8000> by running:

```shell
mkdocs serve
```

The language specification can also be viewed via a provisioned Docker container; no virtual environment is required in this case. Run the Docker container with the following command

```shell
docker compose up -d
```

The language specification can now be viewed at <http://localhost:8106>.

File: README.md [hunk 2]
```diff
@@ -2,12 +2,53 @@
 
 This repository contains the specification documents of the cQASM quantum programming language.
```
Comment (elenbaasc) [inline README.md:3]:
```suggestion
This repository contains the language specification of the cQASM quantum programming language. cQASM stands for common Quantum ASsembly language.
```

File: README.md [hunk 3]
```diff
@@ -1,13 +1,58 @@
...
 
-This repository contains the specification documents of the cQASM quantum programming language. 
+[![License](https://img.shields.io/badge/License-Apache_2.0-blue.svg)](https://opensource.org/licenses/Apache-2.0)
 
-_The specification hosted through GitHub Pages under the following link:_ 
+This repository contains the language specification of the cQASM quantum programming language.
+cQASM stands for *c*ommon *Q*uantum *AS*se*M*bly language.
 
-### [<img src="./docs/_static/book.svg"> cQASM language specification](https://qutech-delft.github.io/cQASM-spec/) 
+## Deployment
```
Comment (elenbaasc) [inline README.md:8]: The Deployment section should indeed be a page confluence. So it should be removed from the README.

As mentioned in a previous comment, the following section can be added to the README, for use by potential contributors:

## Generate the cQASM language specification locally

Clone the repository to your local machine

```shell
git clone https://github.com/QuTech-Delft/cQASM-spec.git
```

We recommend to create a virtual environment (_e.g._, [venv](https://docs.python.org/3/library/venv.html)), run it, and install the requirements

```shell
pip install -r requirements.txt
```

Once the required dependencies have installed, the language specification can be served locally at <http://localhost:8000> by running:

```shell
mkdocs serve
```

The language specification can also be viewed via a provisioned Docker container; no virtual environment is required in this case. Run the Docker container with the following command

```shell
docker compose up -d
```

The language specification can now be viewed at <http://localhost:8106>.

File: README.md [hunk 4]
```diff
@@ -1,13 +1,58 @@
...
+### Using the Docker container
+
+The documentation can also be viewed via the provisioned Docker container.
+Run the docker container with the following command:
+
+```shell
+docker compose up -d
+```
+
+The documentation can now be viewed at <http://localhost:8106>.
```
Comment (elenbaasc) [inline README.md:45]: Move to confluence page.

File: README.md [hunk 5]
```diff
@@ -1,13 +1,58 @@
...
+
+## Documentation
```
Comment (elenbaasc) [inline README.md:47]: We don't have a Documentation section for the cQASM spec, because the specification _is_ the docs.

File: README.md [hunk 6]
```diff
@@ -1,13 +1,58 @@
...
+
+The [cQASM language specification documentation](https://qutech-delft.github.io/cQASM-spec/) is hosted through GitHub Pages.
```
Comment (elenbaasc) [inline README.md:49]: This should be part of the intro (after line 6).
```suggestion
The [cQASM language specification](https://qutech-delft.github.io/cQASM-spec/) is hosted through GitHub Pages.
```

**Review by elenbaasc — COMMENTED** [review]
Some restructuring required (mainly because we use the mkdocs to write the specification, and not a documentation about the specification) and the Deployment should indeed be a confluence page.

File: README.md (hunk 4 above)
Comment (rturrado) [reply README.md:45]: Created: https://qutech-sd.atlassian.net/wiki/spaces/QW/pages/3555950593/Deployment.

File: README.md (hunk 5 above)
Comment (rturrado) [reply README.md:47]: Done.

File: README.md (hunk 6 above)
Comment (rturrado) [reply README.md:49]: Done.

File: README.md (hunk 3 above)
Comment (rturrado) [reply README.md:8]: Done. With:
- some minor tweaks (e.g., adding colons at the end of a phrase), and
- changing the title to `How to generate...`.

Comment (elenbaasc) [reply README.md:8]: I'm not a fan of the ellipsis (`...`), it could be `How to generate` or `Generate the specification locally`

**Review by elenbaasc — APPROVED** [review]
Perfect, thanks for the work!

**Pull request merged**